dagshub/data_engine/model/datasource.py

```diff
@@ -742,6 +749,19 @@ def save_dataset(self, name: str) -> "Datasource":
         copy_with_ds_assigned.load_from_dataset(dataset_name=name, change_query=False)
         return copy_with_ds_assigned
 
+    def _autolog_mlflow(self):
+        if not is_mlflow_installed:
+            return
+        # Run ONLY if there's an active run going on
+        if mlflow.active_run() is None:
+            return
+        source_name = self.source.name
+        with _autolog_counter_lock:
+            counter_value = _autolog_counters[source_name]
+            _autolog_counters[source_name] += 1
+        artifact_name = f"autolog_{source_name}_{counter_value}.dagshub.json"
+        threading.Thread(target=self.log_to_mlflow, kwargs={"artifact_name": artifact_name}).start()
```

Comment:
Would be wise to pass along the active run as a parameter

Reply:
And, are you very sure that `log_to_mlflow` is safe to do async?
The datasource object won't be modified, or otherwise be made into garbage by the time `log_to_mlflow` gets executed?

Reply:
The datasource object is deep copied on every modification, so `log_to_mlflow`is guaranteed to be working with the same object